### PR TITLE
killswitch extendable config format

### PR DESCRIPTION
**Pull request by guliashvili**
Killswitch map of the enabled/disabled switches will be under tables name.

### CODE REVIEW

File: osquery/killswitch/killswitch_plugin.cpp [hunk 1]
```diff
@@ -31,7 +31,17 @@ KillswitchPlugin::parseMapJSON(const std::string& content) {
         "Error parsing the killswitch JSON. Content : " + content);
   }
 
-  for (const auto& keyValue : doc.doc().GetObject()) {
+  const auto table = doc.doc().FindMember("table");
+  if (table == doc.doc().MemberEnd()) {
+    return createError(KillswitchPlugin::ParseMapJSONError::MissingKey,
+                       "killswitch key table containing map was not found");
```
Comment (obelisk) [inline osquery/killswitch/killswitch_plugin.cpp]: Nit, capitalize killswitch

File: osquery/killswitch/killswitch_plugin.cpp [hunk 2]
```diff
@@ -31,7 +31,17 @@ KillswitchPlugin::parseMapJSON(const std::string& content) {
...
+  }
+  if (!table->value.IsObject()) {
+    return createError(KillswitchPlugin::ParseMapJSONError::IncorrectValueType,
+                       "killswitch table value is not an object");
```
Comment (obelisk) [inline osquery/killswitch/killswitch_plugin.cpp]: Same

File: osquery/killswitch/plugins/killswitch_tls.cpp [hunk 3]
```diff
@@ -78,7 +78,12 @@ TLSKillswitchPlugin::refresh() {
   auto it = tree.doc().FindMember("config");
   if (it == tree.doc().MemberEnd()) {
     return createError(KillswitchRefreshablePlugin::RefreshError::ParsingError,
-                       "killswitch member config is not string");
+                       "killswitch member config is missing");
```
Comment (obelisk) [inline osquery/killswitch/plugins/killswitch_tls.cpp]: Same

File: osquery/killswitch/plugins/killswitch_tls.cpp [hunk 4]
```diff
@@ -78,7 +78,12 @@ TLSKillswitchPlugin::refresh() {
...
+  }
+
+  if (!it->value.IsString()) {
+    return createError(KillswitchRefreshablePlugin::RefreshError::ParsingError,
+                       "killswitch member config is not a string");
```
Comment (obelisk) [inline osquery/killswitch/plugins/killswitch_tls.cpp]: Same

**Comment by guliashvili** [comment]
Thanks, updated messages everywhere in the killswitch @obelisk

**Comment by guliashvili** [comment]
@obelisk Added tests for invalid killswitch config.

**Pull request merged**